dali/imgcodec/decoders/decoder_test_helper.h

```diff
@@ -39,54 +43,98 @@ namespace test {
+      auto tv = output_.cpu()[0];
+      SampleView<CPUBackend> view(tv.data, tv.shape, type2id<OutputType>::value);
+      DecodeResult decode_result = Decoder()->Decode(view, src, opts, roi);
+      EXPECT_TRUE(decode_result.success);
+      return tv;
+    } else {  // GPU
+      auto tv = output_.gpu()[0];
+      SampleView<GPUBackend> view(tv.data, tv.shape, type2id<OutputType>::value);
+      auto stream = CUDAStreamPool::instance().Get(GetDeviceId());
+      auto decode_result = Decoder()->Decode(stream, view, src, opts, roi);
+      EXPECT_TRUE(decode_result.success);
+      StreamSync(stream);
+      return output_.cpu()[0];
     }
-    result.Resize(shape, type2id<OutputType>::value);
+  }
 
-    SampleView<CPUBackend> view(result.raw_mutable_data(), result.shape(), result.type());
-    DecodeResult decode_result = Decoder()->Decode(view, src, opts, roi);
-    EXPECT_TRUE(decode_result.success);
+  /**
+   * @brief Decodes a batch of images, invoking the batch version of ImageDecoder::Decode
+   */
+  TensorListView<StorageCPU, OutputType> Decode(cspan<ImageSource *> in,
+                                         const DecodeParams &opts = {},
+                                         cspan<ROI> rois = {}) {
+    int n = in.size();
+    std::vector<TensorShape<>> shape(n);
+    std::vector<Tensor<CPUBackend>> result(n);
 
-    return result;
+    for (int i = 0; i < n; i++) {
+      EXPECT_TRUE(Parser()->CanParse(in[i]));
+      EXPECT_TRUE(Decoder()->CanDecode(in[i], opts));
+      ImageInfo info = Parser()->GetInfo(in[i]);
+      shape[i] = AdjustToRoi(info.shape, rois.empty() ? ROI{} : rois[i]);
+      result[i].Resize(shape[i], type2id<OutputType>::value);
+    }
+
+    output_.reshape(TensorListShape{shape});
+
+    if (GetDeviceId() == CPU_ONLY_DEVICE_ID) {
+      auto tlv = output_.cpu();
+      std::vector<SampleView<CPUBackend>> view(n);
+      for (int i = 0; i < n; i++)
+        view[i] = {tlv[i].data, tlv[i].shape, type2id<OutputType>::value};
+      auto res = Decoder()->Decode(make_span(view), in, opts, rois);
+      for (auto decode_result : res)
+        EXPECT_TRUE(decode_result.success);
+      return tlv;
+    } else {  // GPU
+      auto tlv = output_.gpu();
+      std::vector<SampleView<GPUBackend>> view(n);
+      for (int i = 0; i < n; i++)
+        view[i] = {tlv[i].data, tlv[i].shape, type2id<OutputType>::value};
+      auto stream = CUDAStreamPool::instance().Get(GetDeviceId());
+      auto res = Decoder()->Decode(stream, make_span(view), in, opts, rois);
+      for (auto decode_result : res)
+        EXPECT_TRUE(decode_result.success);
+      StreamSync(stream);
+      return output_.cpu();
+    }
   }
 
   /**
   * @brief Checks if two tensors are equal.
   */
-  void AssertEqual(const Tensor<CPUBackend> &img, const Tensor<CPUBackend> &ref) {
-    Check(view<const OutputType>(img), view<const OutputType>(ref));
+  void AssertEqual(const TensorView<StorageCPU, OutputType> &img, const Tensor<CPUBackend> &ref) {
```

Comment:
I'd recommend making `ref` a tensor-view too. It's a more flexible interface.

Reply:
I might just add it as another overload, as this version is pretty convenient to use with `NumpyDecoderTestBase::ReadReference`. Or maybe add it to `NumpyDecoderTestBase` itself?

Reply:
I mean, it's easy to do `view<T>(ref)` when calling. My point was that having a tensor view in the interface would allow for more flexible usage, as the tensor view can be also constructed from any memory location.